src/core/lib/security/security_connector/load_system_roots_windows.cc

```diff
@@ -0,0 +1,90 @@
+//
+//
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#include <grpc/support/port_platform.h>
+
+#if defined(GPR_WINDOWS)
+
+#pragma comment(lib, "crypt32")
+
+#include <esent.h>
+#include <wincrypt.h>
+
+#include <vector>
+
+#include <grpc/slice.h>
+#include <grpc/slice_buffer.h>
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+
+#include "src/core/lib/gpr/useful.h"
+#include "src/core/lib/security/security_connector/load_system_roots.h"
+
+namespace grpc_core {
+namespace {
+
+std::string utf8Encode(const std::wstring& wstr) {
+  if (wstr.empty()) return std::string();
+
+  int sizeNeeded = WideCharToMultiByte(CP_UTF8, 0, &wstr[0], (int)wstr.size(),
+                                       NULL, 0, NULL, NULL);
+  std::string strTo(sizeNeeded, 0);
+  WideCharToMultiByte(CP_UTF8, 0, &wstr[0], (int)wstr.size(), &strTo[0],
+                      sizeNeeded, NULL, NULL);
+  return strTo;
+}
+
+}  // namespace
+
+grpc_slice LoadSystemRootCerts() {
```

Comment:
Not directly related to this PR, but at some point it would be good to change this function to return `grpc_core::Slice` instead of `grpc_slice`.  The former is a C++ type with much more idiomatic semantics.